enzyme/Enzyme/GradientUtils.cpp

```diff
@@ -9182,15 +9185,21 @@ llvm::CallInst *freeKnownAllocation(llvm::IRBuilder<> &builder,
     freefunc = LibFunc_ZdaPv;
     break;
 
-  case LibFunc_msvc_new_int:               // new(unsigned int);
-  case LibFunc_msvc_new_int_nothrow:       // new(unsigned int, nothrow);
   case LibFunc_msvc_new_longlong:          // new(unsigned long long);
   case LibFunc_msvc_new_longlong_nothrow:  // new(unsigned long long, nothrow);
-  case LibFunc_msvc_new_array_int:         // new[](unsigned int);
-  case LibFunc_msvc_new_array_int_nothrow: // new[](unsigned int, nothrow);
+    freefunc = LibFunc_msvc_delete_ptr64_longlong;
+    break;
+
   case LibFunc_msvc_new_array_longlong:    // new[](unsigned long long);
   case LibFunc_msvc_new_array_longlong_nothrow: // new[](unsigned long long,
                                                 // nothrow);
+    freefunc = LibFunc_msvc_delete_array_ptr64_longlong;
+    break;
```

Comment:
Add some tests?

Reply:
Where would these tests go?

Reply:
Inside the corresponding subfolder here https://github.com/EnzymeAD/Enzyme/tree/main/enzyme/test/Enzyme 

e.g. a reverse mode test in ReverseMode/ etc

Reply:
Do you have a CI job targeting the MS x64 ABI?

Reply:
You shouldn’t need a CI job, just the LLVM emitted by a code with the MS x86 ABI

Reply:
I see. You are asking for the thing I feared you might ask for. So that I might make the uber MS test case, can you address the first question up there, namely, can we get `metaString->contains("enzyme_")` in Enzyme.cpp? Unfortunately, the `enzyme_dup, enzyme_const, ...` are mangled by `LLVMEnzyme.dll` here in the `metaString`. They are not mangled by `ClangEnzyme.dll`.

Reply:
Would an extern "C" int enzyme_int; work?

I'm also fine supporting the mangled names, but we may want to not just do includes but instead unmangled name equals

@ZuseZ4 set that up previously for activity analysis so many be able to help.

Reply:
extern "C" seems to work fine, so it is just the c++ test code that needs to change. I'll do that too.

Reply:
I just added a file on which you can run opt and hit most of these changes.